src/components/external-source/ExternalSourceManager.svelte

```diff
@@ -0,0 +1,1050 @@
+                  type="search"
+                  bind:value={filterExpression}
+                  placeholder="Filter External Sources"
+                  class="st-input"
+                />
+              </Input>
+            </div>
+          </div>
+        </slot>
+        <slot name="right">
+          <button
+            name="create-groups-or-types"
+            class="st-button secondary"
+            on:click|stopPropagation={onCreateGroupsOrTypes}
+            use:tooltip={{
+              content: 'Create derivation groups, external source types, or external event types.',
+              placement: 'top',
+            }}
+          >
+            Create New Groups or Types
+          </button>
+          <button
+            name="manage-groups-or-types"
+            class="st-button active"
+            on:click|stopPropagation={onManageGroupsAndTypes}
+            use:tooltip={{
+              content:
+                'Manage and inspect existing derivation groups, external source types, and external event types.',
+              placement: 'top',
+            }}
+          >
+            Manage Groups and Types
+          </button>
+        </slot>
+      </svelte:fragment>
+      <svelte:fragment slot="body">
+        {#if $externalSources.length}
+          <BulkActionDataGrid
+            {columnDefs}
+            {hasDeletePermission}
+            singleItemDisplayText="External Source"
+            pluralItemDisplayText="External Source"
+            {filterExpression}
+            items={filteredExternalSources}
+            {user}
+            getRowId={getRowIdExternalSourceSlim}
+            on:rowClicked={({ detail }) => selectSource(detail.data)}
+            on:bulkDeleteItems={({ detail }) => onDeleteExternalSource(detail)}
+            bind:selectedItemId={selectedSourceId}
+          />
+        {:else}
+          <p>No external sources matching the selected external source type(s).</p>
+        {/if}
+      </svelte:fragment>
+    </Panel>
+
+    {#if selectedSource}
+      <CssGridGutter track={1} type="row" />
+
+      <!-- External Event Table/Timeline -->
+      <Panel padBody={true}>
+        <svelte:fragment slot="header">
+          <slot name="left">
+            <SectionTitle><Balloon slot="icon" />External Events</SectionTitle>
+            <div class="filter">
+              <div class="timeline-editor-layer-filter">
+                <Input>
+                  <input
+                    bind:value={externalEventsTableFilterString}
+                    autocomplete="off"
+                    class="st-input w-100"
+                    name="filter-ee"
+                    placeholder={'Filter external events'}
+                  />
+                </Input>
+              </div>
+            </div>
+          </slot>
+        </svelte:fragment>
+        <svelte:fragment slot="body">
+          {#if selectedSource}
+            <div id="external-event-table">
+              <ExternalEventsTable
```

Comment:
Instead of manually handling the filtering could this instead pass in a `filterExpression` as is done with the External Sources table? This may require passing the prop down into the ExternalEventsTable and also through the SingleActionDataGrid component.

Reply:
Benefit is that both filter inputs on this page will behave the same. Right now the top filters on all columns whereas the bottom only filters on the key.

Reply:
This commit ([update ExternalEventsTable to BulkActionDataGrid](https://github.com/NASA-AMMOS/aerie-ui/pull/1396/commits/2ac94803616ffec6c4d7867269eb9058b5a9d851)) handles this, though I question whether making both of these `BulkActionDataGrid`s, the only type with a built in `filterExpression`, is a good idea as multi-selection and a greyed out option to delete don't really make sense here.

Reply:
Could just pass in `filterExpression` to `SingleActionDataGrid` and have that component pass it down to `DataGrid` since it is really just a pass through to AGGrid?

Reply:
Addressed/replaced in [simplify ExternalEventsTable.svelte](https://github.com/NASA-AMMOS/aerie-ui/pull/1396/commits/638bc50ccab269f9d8400bcffd2c6f8d9a8c3961).